src/components/checkbox/checkbox.spec.ts

```diff
@@ -0,0 +1,713 @@
+
+        controller.isIndeterminate = false;
+        fixture.detectChanges();
+        expect(el.nativeElement.className).not.toContain(
+            'md-checkbox-anim-unchecked-indeterminate');
+        expect(el.nativeElement.className).toContain('md-checkbox-anim-indeterminate-unchecked');
+      });
+
+      it('applies a transition class when going from checked -> indeterminate', function() {
+        let el = fixture.debugElement.query(By.css('.md-checkbox'));
+
+        controller.isChecked = true;
+        fixture.detectChanges();
+
+        controller.isIndeterminate = true;
+        fixture.detectChanges();
+        expect(el.nativeElement.className).not.toContain('md-checkbox-anim-unchecked-checked');
+        expect(el.nativeElement.className).toContain('md-checkbox-anim-checked-indeterminate');
+      });
+
+      it('applies a transition class when going from indeterminate -> checked', function() {
+        let el = fixture.debugElement.query(By.css('.md-checkbox'));
+
+        controller.isIndeterminate = true;
+        fixture.detectChanges();
+
+        controller.isChecked = true;
+        fixture.detectChanges();
+        expect(el.nativeElement.className).not.toContain(
+            'md-checkbox-anim-unchecked-indeterminate');
+        expect(el.nativeElement.className).toContain('md-checkbox-anim-indeterminate-checked');
+      });
+
+      it('does not apply any transition classes when there is nothing to transition', function() {
+        let el = fixture.debugElement.query(By.css('.md-checkbox'));
+
+        controller.isChecked = controller.isChecked;
+        fixture.detectChanges();
+        expect(el.nativeElement.className).not.toMatch(/^md\-checkbox\-anim/g);
+
+        controller.isIndeterminate = controller.isIndeterminate;
+        fixture.detectChanges();
+        expect(el.nativeElement.className).not.toMatch(/^md\-checkbox\-anim/g);
+      });
+
+      it('does not apply any transition classes when the component is initialized', function() {
+        let el = fixture.debugElement.query(By.css('.md-checkbox'));
+        expect(el.nativeElement.className).not.toMatch(/^md\-checkbox\-anim/g);
+      });
+    });
+  });
+}
+
+function click(el: DebugElement, fixture: ComponentFixture) {
+  let clickEvent = new Event('click');
+  spyOn(clickEvent, 'stopPropagation').and.callThrough();
+  el.nativeElement.dispatchEvent(clickEvent);
+  fixture.detectChanges()
+  return clickEvent;
+}
+
+function keyup(el: DebugElement, key: string, fixture: ComponentFixture) {
+  let kbdEvent = new KeyboardEvent('keyup');
+  // Hack DOM Level 3 Events "key" prop into keyboard event.
+  Object.defineProperty(kbdEvent, 'key', {
+    value: ' ',
+    enumerable: false,
+    writable: false,
+    configurable: true
+  });
+  spyOn(kbdEvent, 'stopPropagation').and.callThrough();
+  el.nativeElement.dispatchEvent(kbdEvent);
+  fixture.detectChanges();
+  return kbdEvent;
+}
```

Comment:
When we start running unit tests in Dart these definitely aren't going to work, but we can address that in a follow-up. Could you leave a TODO?